authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt

```diff
@@ -0,0 +1,195 @@
+package com.bitwarden.authenticatorbridge.manager
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.os.IBinder
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ProcessLifecycleOwner
+import com.bitwarden.authenticatorbridge.IAuthenticatorBridgeService
+import com.bitwarden.authenticatorbridge.manager.model.AccountSyncState
+import com.bitwarden.authenticatorbridge.manager.model.AuthenticatorBridgeConnectionType
+import com.bitwarden.authenticatorbridge.manager.util.toPackageName
+import com.bitwarden.authenticatorbridge.model.EncryptedSharedAccountData
+import com.bitwarden.authenticatorbridge.provider.AuthenticatorBridgeCallbackProvider
+import com.bitwarden.authenticatorbridge.provider.StubAuthenticatorBridgeCallbackProvider
+import com.bitwarden.authenticatorbridge.provider.SymmetricKeyStorageProvider
+import com.bitwarden.authenticatorbridge.util.decrypt
+import com.bitwarden.authenticatorbridge.util.toFingerprint
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+private const val AUTHENTICATOR_BRIDGE_SERVICE_CLASS =
+    "com.x8bit.bitwarden.data.platform.service.AuthenticatorBridgeService"
+
+/**
+ * Default implementation of [AuthenticatorBridgeManager].
+ *
+ * @param context The Context that will be used to bind to AuthenticatorBridgeService.
+ * @param connectionType Specifies which build variant to connect to.
+ * @param symmetricKeyStorageProvider Provides access to local storage of the symmetric encryption
+ * key.
+ * @param callbackProvider Provides a way to construct a service callback that can be mocked in
+ * tests.
+ * @param processLifecycleOwner Lifecycle owner that is used to listen for start/stop
+ * lifecycle events.
+ */
+internal class AuthenticatorBridgeManagerImpl(
+    private val connectionType: AuthenticatorBridgeConnectionType,
+    private val symmetricKeyStorageProvider: SymmetricKeyStorageProvider,
+    callbackProvider: AuthenticatorBridgeCallbackProvider = StubAuthenticatorBridgeCallbackProvider(),
+    context: Context,
+    processLifecycleOwner: LifecycleOwner = ProcessLifecycleOwner.get(),
+) : AuthenticatorBridgeManager {
+
+    private val applicationContext = context.applicationContext
+
+    /**
+     * Main AuthenticatorBridgeService access point.
+     */
+    private var bridgeService: IAuthenticatorBridgeService? = null
+
+    /**
+     * Internal state of [accountSyncStateFlow].
+     */
+    private val mutableSharedAccountsStateFlow =
+        MutableStateFlow<AccountSyncState>(AccountSyncState.Loading)
```

Comment:
So we can determine in the `init` block if this should be `Loading` because the BW app is installed and we have a shot at binding to it, or something more like `ApplicationNotFound` if the BW app doesn't exist.

Reply:
I added a check during the `bindService` call as well. Basically, if the app doesn't exist, don't bind. 

1df5c36e3738da75adcfea5df5bc5263aab90b80